### PR TITLE
Makefile Refactor and gppkg build script

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. refactored the Makefile, remove unnecessary make targets. 
2. The gppkg script also refactored and now support OSS gpdb.
3. Add complier flag to adjust docker API version

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,9 +51,17 @@ endif
 
 PLCONTAINERDIR = $(DESTDIR)$(datadir)/plcontainer
 
+override CFLAGS += -Werror
+
+# detected the docker API version
+OS_VERSION=$(shell cat /etc/redhat-release | sed s/.*release\ // | sed s/\ .*// | awk -F '.' '{print $$1}' )
+ifneq ($(OS_VERSION), 7)
+  override CFLAGS +=  -DDOCKER_API_LOW
+endif
```
Comment (paul-guo-) [inline Makefile]: I'd suggest warn if it is not redhat distribution.

File: package/Makefile [hunk 2]
```diff
@@ -5,14 +5,14 @@ include $(PGXS)
 
 include ../release.mk
 
-OS=$(word 1,$(subst _, ,$(BLD_ARCH)))
+OS_PL=$(shell cat /etc/redhat-release | sed s/.*release\ // | sed s/\ .*// | awk -F '.' '{print $$1}' )
```
Comment (paul-guo-) [inline package/Makefile]: Maybe MAJOR_OS is better?

File: src/plc_docker_curl_api.c [hunk 3]
```diff
@@ -24,8 +24,11 @@
 static char *plc_docker_socket = "/var/run/docker.sock";
 
 // URL prefix specifies Docker API version
-static char *plc_docker_url_prefix = "http:/v1.21";
-
+#ifdef DOCKER_API_LOW
+    static char *plc_docker_url_prefix = "http:/v1.19";
+#else
+    static char *plc_docker_url_prefix = "http:/v1.21";
+#endif
```
Comment (paul-guo-) [inline src/plc_docker_curl_api.c]: We need to modify for the non-curl version also.

**Comment by Eulerizeit** [comment]
@ivannovick do we need to worry about Ubuntu or SuSE 12 in the near future?

**Comment by davecramer** [comment]
@Eulerizeit why limit it at all ? it works fine as is

File: Makefile (hunk 1 above)
Comment (HaozhouWang) [inline Makefile]: Thanks for your review, changed to only for centos 6 docker version

File: package/Makefile (hunk 2 above)
Comment (HaozhouWang) [inline package/Makefile]: Thanks for your review, changed to MAJOR_OS

File: src/plc_docker_curl_api.c (hunk 3 above)
Comment (HaozhouWang) [inline src/plc_docker_curl_api.c]: Oh sorry, this is my fault. I will update the non-curl version as well.

**Comment by HaozhouWang** [comment]
We changed the Makefile to use lower version docker API for centos 6. If we need support Ubuntu or SuSE12, we can open a new PR to update the Makefile to detect the OS version and docker API version in that OS.

**Comment by HaozhouWang** [comment]
@paul-guo- Yes, exactly.
We currently build the plcontainer.so firstly, and then clean the  src/common/*.o, and build the clients.

**Comment by stanlyxiang** [comment]
question: do we need to update license header in our pr instead of a separate pr to update all  ? 
@HaozhouWang @paul-guo-

**Comment by paul-guo-** [comment]
Agree. In the long run, we could generate docker version constants for various OSes. If we have more and more dynamic variables, we could consider add configure.

**Comment by HaozhouWang** [comment]
@stanlyxiang +1 to update in our pr

**Pull request merged**